src/common/src/types/chrono_wrapper.rs

```diff
@@ -211,3 +241,33 @@ impl TryFrom<NaiveDateWrapper> for NaiveDateTimeWrapper {
         Ok(NaiveDateTimeWrapper::new(date.0.and_hms(0, 0, 0)))
     }
 }
+
+/// return the days of the `year-month`
+fn get_mouth_days(year: i32, month: usize) -> i32 {
+    if is_leap_year(year) {
+        LEAP_DAYS[month]
+    } else {
+        NORMAL_DAYS[month]
+    }
+}
+
+fn is_leap_year(year: i32) -> bool {
+    year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)
+}
+
+pub trait CheckedAddAssign<T> {
+    fn checked_add_assign(&mut self, rhs: T) -> Result<()>;
+}
+
+impl CheckedAddAssign<i32> for i32 {
+    fn checked_add_assign(&mut self, rhs: i32) -> Result<()> {
+        *self += rhs;
+        Ok(())
+    }
+}
```

Comment:
* Move to `types/mod.rs` or a new `types/ops.rs`
* `impl<T: num_traits::CheckedAdd> CheckedAdd<T> for T` so that we have our owned `CheckedAdd` ready for any types that already impl `num_traits::CheckedAdd` in addition to i32.
* Given the actual impl is `checked_add`, let's just write `cur = cur.checked_add(step)` in generate series executor.